cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,104 @@
 
-# Steps
+The following short guidelines can be used as a quick reference to protect the forgot password or MFA service:
 
-## Step 1) Gather Identity Data or Security Questions
+- **Unified user message, whether the user exists or not.**
+- **Ensure that the time taken for the user response message is uniform.**
+- **Use a side-channel to communicate the method to reset their password or MFA.**
+- **For critical services and applications, MFA should be used.**
 
-The first page of a secure Forgot Password feature asks the user for multiple pieces of hard data that should have been previously collected (generally when the user first registers). 
+## Content
 
-Steps for this are detailed in the identity section of the [Choosing and Using Security Questions Cheat Sheet](Choosing_and_Using_Security_Questions_Cheat_Sheet.md#choosing-security-questions).
+- [Forgot Password and MFA Cheat Sheet](#forgot-password-and-mfa-cheat-sheet)
+  * [Introduction](#introduction)
+  * [Content](#content)
+  * [Methods](#methods)
+  * [Methods Implementation](#methods-implementation)
+    + [One Time Password](#one-time-password)
+    + [URL Tokens](#url-tokens)
+    + [Security Questions](#security-questions)
+    + [Backup Codes](#backup-codes)
+  * [Operational Tasks](#operational-tasks)
 
-At a minimum, you should have collected some data that will allow you to send the password reset information to some out-of-band side-channel, such as a (possibly different) email address or an SMS text number, etc. to be used in Step 3.
+## Methods
 
-## Step 2) Verify Security Questions
+In order to allow a user to request a password or MFA reset, you will need to have some way to identify the user, or a mean to reach out to them through a side-channel.
 
-After the form on Step 1 is submitted, the application verifies that each piece of data is correct for the given username. If anything is incorrect, or if the username is not recognized, the second page displays a generic error message such as *Sorry, invalid data*. 
+This can be done through any of the following methods:
 
-If all submitted data is correct, Step 2 should display at least two of the user's pre-established personal security questions, along with input fields for the answers. It's important that the answer fields are part of a single HTML form.
+- One Time Password (OTP).
+- URL tokens.
+- Security Questions.
+- Offline backup codes.
 
-Do not provide a drop-down list for the user to select the questions he wants to answer. Avoid sending the username as a parameter (hidden or otherwise) when the form on this page is submitted. The username should be stored in the server-side session where it can be retrieved as needed.
+> These methods can be used together, and it is recommended most of the times. Make sure that the user always have a mean to recover their account.
```

Comment:
Is this meant to be a quote? It's also got a couple small issues. I'd rephrase as the below:

> These can be used together and many times it is recommended to do so. No matter what you must ensure that a user always has a way to recover their account.

Reply:
Quotes are used as Notes. They step out of the context of the thing being written. The sentence is improved as well, thank you